### PR TITLE
Object detection transforms tooltips and local result file generation

### PR DESCRIPTION
**Pull request by Dashrekker**
Added local generation of detection_results.csv and tooltips for detection transforms. Also converted the transforms ui to the layer choice used for ml models.
image

### CODE REVIEW

**Review by farisdurrani — APPROVED** [review]

**Review by farisdurrani — APPROVED** [review]

File: backend/common/constants.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "./frontend/playground-frontend/src/backend_outputs/visualization_output/confusion_matrix.png"
 )
 CLASSICAL_ML_RESULT_CSV_PATH = f"ml_results.csv"
+IMAGE_DETECTION_RESULT_CSV_PATH = f"detection_results.csv"
```
Comment (karkir0003) [inline backend/common/constants.py:24]: where is the code that writes object detection results to this file path or is this only on download? @Dashrekker

File: backend/dl/detection.py [hunk 2]
```diff
@@ -66,4 +67,17 @@ def detection_img_drive(IMAGE_UPLOAD_FOLDER, detection_type, problem_type, trans
...
+    if label_set:
+        backend_dir = (
+            ""
+            if (os.getcwd()).split("\\")[-1].split("/")[-1] == "backend"
+            else "backend"
+            if ("backend" in os.listdir(os.getcwd()))
+            else "../backend"
+        )
+        keys = label_set[0].keys()
+        with open(os.path.join(backend_dir, "detection_results.csv"), 'w', newline = '') as f:
```
Comment (karkir0003) [inline backend/dl/detection.py:79]: any particular reason this logic lives in `transform_image()` function. Might want to have a separate helper function that writes the object detection results to the appropriate backend path @Dashrekker

**Review by karkir0003 — CHANGES_REQUESTED** [review]
address comments

File: backend/dl/detection.py (hunk 2 above)
Comment (Dashrekker) [reply backend/dl/detection.py:69]: just added the separate helper function as write_to_csv

File: backend/common/constants.py (hunk 1 above)
Comment (Dashrekker) [reply backend/common/constants.py:24]: the code is in the write_to_csv function in detection.py

**Review by karkir0003 — APPROVED** [review]

**Comment by farisdurrani** [comment]
PR approved. @Dashrekker feel free to merge it in

**Pull request merged**